tests/spec/core/highlight-spec.js

```diff
@@ -76,4 +76,20 @@ describe("Core — Highlight", () => {
     const pre = doc.getElementById("test");
     expect(pre.querySelectorAll("span[class^=hljs-]").length).toBe(0);
   });
+
+  it("checks if <pre> content is warpped in <code>", async () => {
+    const ops = {
+      config: makeBasicConfig(),
+      body: `${makeDefaultBody()}<section>
+          <pre>
+            function foo() {
+              alert('foo');
+            }
+          </pre>
```

Comment:
don't forget to also test:

```JS
<p><dfn>a link</dfn></p>
<pre><!--empty--></pre>
<pre>
   <a>a link</a>
  <code>
    function one(){}
  </code>
  this function(){} is not highlighted. 
  <code class="http">
     Header: Test
  </code>
</pre>

```

Reply:
Yes, as of now I haven't added the case when there is `<code>` inside `<pre>` so, I'll add this when I handle this case in `highlight.js`, so this test is just when there is no `<code>` inside `<pre>`.

Reply:
Ok, that's fine - so make sure you always check the default condition too (i.e., when there is a `<code>`):

```JS
<pre><code lang="http">test</code></pre>
```

To make sure that it doesn't get double wrapped by code, and that the language remains intact, etc.

Reply:
I thought `lang` is for spoken languages?

Reply:
sorry, yes... that should be class="http".

Reply:
fixed above.